test/e2e/garbage_collector.go

```diff
@@ -132,6 +132,15 @@ func verifyRemainingDeploymentsAndReplicaSets(
 		ret = false
 		By(fmt.Sprintf("expected %d Deploymentss, got %d Deployments", deploymentNum, len(deployments.Items)))
 	}
+	pods, err := clientSet.CoreV1().Pods(f.Namespace.Name).List(metav1.ListOptions{})
+	if err != nil {
+		return false, fmt.Errorf("Failed to list pods: %v", err)
+	}
+	if len(rsNum) == 0 && len(pods) != podNum {
```

Comment:
@krmayankk 
`len(rsNum)` -  is type `int` - that can't work, right?
`len(pods)` - is type `*"k8s.io/kubernetes/pkg/api/v1".PodList` - I don't think this can work either 

I would suggest something like: `if rsNum == 0 && pods.Size() != podNum {`